### PR TITLE
Separate sql changes into own files and auto generate the full script.

### PR DESCRIPTION
**Pull request by smithago**
## Description
Improves developer productivity with Sql changes.
Both making and code reviewing changes.
Diff files are still manually generated. 

## Related issues
[AB#85587](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/85587)

## Testing
Build works

### CODE REVIEW

File: src/Microsoft.Health.Dicom.SqlServer/Microsoft.Health.Dicom.SqlServer.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="Microsoft.Health.Extensions.BuildTimeCodeGenerator" Version="$(HealthcareSharedPackageVersion)" PrivateAssets="All" />
     <PackageReference Include="Microsoft.Health.SqlServer" Version="$(HealthcareSharedPackageVersion)" />
     <PackageReference Include="Microsoft.Health.SqlServer.Api" Version="$(HealthcareSharedPackageVersion)" />
+    <PackageReference Include="Microsoft.Health.Tools.Sql.Tasks" Version="1.0.0" />
```
Comment (smithago) [inline src/Microsoft.Health.Dicom.SqlServer/Microsoft.Health.Dicom.SqlServer.csproj:21]: This is manually uploaded for now to unblock.  I will comeback and fix the version after that work is done

**Review by jovinson-ms — APPROVED** [review]
Tested creating a new version by cloning the latest version - unit tests pass including validating the schema.

**Review by jovinson-ms — APPROVED** [review]
Since the generated schema files have some formatting differences compared to the hand-created files, the integration tests covering schema upgrade fail. The more difficult one is the comparison between the new schema file (e.g. `5.sql`) and the diff file (`5.diff.sql`) because it does a very strict comparison, even finding changed comments.

File: src/Microsoft.Health.Dicom.SqlServer/Microsoft.Health.Dicom.SqlServer.csproj [hunk 2]
```diff
@@ -132,13 +133,14 @@
     <SqlScript Include="Features\Schema\Sql\Scripts\FullText.sql" />
   </ItemGroup>
   
+  <!-- Properties used by sql task to generate full script-->
   <PropertyGroup>
     <LatestSchemaVersion>5</LatestSchemaVersion>
     <IntermediateOutputPath>$(MSBuildThisFileDirectory)obj\$(Configuration)</IntermediateOutputPath>
```
Comment (jovinson-ms) [inline src/Microsoft.Health.Dicom.SqlServer/Microsoft.Health.Dicom.SqlServer.csproj:139]: Is there a configuration option to leave formatting alone? The current task is stripping out comments and reducing whitespace (for example, consolidating the parameters onto one line) in stored procedures. This is causing the schema comparison test to fail.

Comment (smithago) [reply src/Microsoft.Health.Dicom.SqlServer/Microsoft.Health.Dicom.SqlServer.csproj:139]: Yup, the diff of Prc is literal Bodyscript compare. I am skipping that compare for now. We will still

**Pull request merged**